nemo/utils/nemo_logging.py

```diff
@@ -88,7 +88,17 @@ def _define_logger(self):
                 self._logger = _logging.getLogger("nemo_logger")
                 # By default, silence all loggers except the logger for rank 0
                 self.remove_stream_handlers()
-                if get_envint("RANK", 0) == 0:
+                if get_envbool(NEMO_ENV_VARNAME_TESTING, False):
+                    old_factory = _logging.getLogRecordFactory()
+
+                    def record_factory(*args, **kwargs):
+                        record = old_factory(*args, **kwargs)
+                        record.rank = get_envint("RANK", 0)
+                        return record
+
+                    _logging.setLogRecordFactory(record_factory)
+                    self.add_stream_handlers(formatter=DebugNeMoFormatter)
+                elif get_envint("RANK", 0) == 0:
```

Comment:
who sets this env variable? distributed launcher? Slurm set LOCAL_RANK, I thought distributed launcher sets that too. Shouldn't this be called LOCAL_RANK?

Reply:
To be honest, I'm not sure, but that was how we made sure that only rank 0 was printing to screen. So the behaviour is the same as what we have in master